scripts/ImportExportTest.sh

```diff
@@ -0,0 +1,276 @@
+    READLINK=greadlink
+fi
+REPO_ROOT=$(${READLINK} -f "$(dirname "$0")"/..)
+SOLIDITY_BUILD_DIR=${SOLIDITY_BUILD_DIR:-${REPO_ROOT}/build}
+SOLC=${SOLIDITY_BUILD_DIR}/solc/solc
+SPLITSOURCES=${REPO_ROOT}/scripts/splitSources.py
+
+# shellcheck disable=SC1090,SC1091
+source "${REPO_ROOT}/scripts/common.sh"
+
+SYNTAXTESTS_DIR="${REPO_ROOT}/test/libsolidity/syntaxTests"
+SEMANTICTESTS_DIR="${REPO_ROOT}/test/libsolidity/semanticTests"
+ASTJSONTESTS_DIR="${REPO_ROOT}/test/libsolidity/ASTJSON"
+
+FAILED=0
+UNCOMPILABLE=0
+TESTED=0
+
+if [[ "$(find . -maxdepth 0 -type d -empty)" == "" ]]; then
+    fail "Test directory not empty. Skipping!"
+fi
+
+function Ast_ImportExportEquivalence
+{
+    local sol_file="$1"
+    local input_files="$2"
+    # save exported json as expected result (silently)
+    $SOLC --combined-json ast --pretty-json --json-indent 4 "${input_files}" > expected.json 2> /dev/null
+    # import it, and export it again as obtained result (silently)
+    if ! $SOLC --import-ast --combined-json ast --pretty-json --json-indent 4 expected.json > obtained.json 2> stderr.txt
+    then
+        # For investigating, use exit 1 here so the script stops at the
+        # first failing test
+        # exit 1
+        FAILED=$((FAILED + 1))
+        printError -e "ERROR: AST reimport failed for input file $sol_file"
+        printError
+        printError "Compiler stderr:"
+        cat ./stderr.txt >&2
+        printError
+        printError "Compiler stdout:"
+        cat ./obtained.json >&2
+        return 1
+    fi
+    if ! diff_files expected.json obtained.json
+    then
+        FAILED=$((FAILED + 1))
+    fi
+    TESTED=$((TESTED + 1))
+    rm expected.json obtained.json
+    rm -f stderr.txt
+}
+
+function JsonEvmAsm_ImportExportEquivalence
+{
+    local sol_file="$1"
+    local input_files="$2"
+    local outputs=( "asm" "bin" "bin-runtime" "opcodes" "srcmap" "srcmap-runtime" )
+    local _TESTED=1
+    if ! "${SOLC}" --combined-json "$(IFS=, ; echo "${outputs[*]}")" --pretty-json --json-indent 4 "${input_files}" > expected.json 2> expected.error
+    then
+        printError
+        printError "$sol_file"
+        cat expected.error >&2
+        UNCOMPILABLE=$((UNCOMPILABLE + 1))
+        return 0
+    fi
+
+    # Note that we have some test files, that only consists of free functions.
+    # Those files doesn't define any contracts, so the resulting json does not define any
+    # keys. In this case `jq` returns an error like `jq: error: null (null) has no keys`
+    # to not get spammed by these errors, errors are redirected to /dev/null.
+    for contract in $(jq '.contracts | keys | .[]' expected.json 2> /dev/null)
+    do
+        for output in "${outputs[@]}"
+        do
+            jq --raw-output ".contracts.${contract}.\"${output}\"" expected.json > "expected.${output}.json"
+        done
+
+        assembly=$(cat expected.asm.json)
+        if [ "$assembly" != "" ] && [ "$assembly" != "null" ]
+        then
```

Comment:
That's really good to prevent unneeded indentation. I put `[ $assembly != "" && $assembly != "null" ]] || continue` here - because we are in the loop that we still want to process.